docs/config.rst

```diff
@@ -1,205 +1,431 @@
 Configuration
```

Comment:
Didn't really review much of this file. This seems mostly like copy/paste and reformatting from the README?